src/main/java/org/traccar/protocol/GatorProtocolDecoder.java

```diff
@@ -120,17 +176,99 @@ protected Object decode(
 
             position.set(Position.KEY_ODOMETER, buf.readUnsignedInt());
 
+            if (type == MSG_ALARM_DATA || type  == MSG_POSITION_DATA) {
+                int temperatureSignPosition = buf.readUnsignedByte();
+                int temperaturePosition = buf.readUnsignedByte();
+
+                if (temperaturePosition != 0xFF) {
+                    if (temperatureSignPosition == 0x00) {
+                        position.set(Position.PREFIX_TEMP + 1, temperaturePosition);
+                    } else if (temperatureSignPosition == 0x01) {
+                        position.set(Position.PREFIX_TEMP + 1, -temperaturePosition);
+                    }
+                }
+            }
+            if (type == MSG_ALARM_DATA) {
+                int alarmDataByte1 = buf.readUnsignedByte();
+                int alarmDataByte2 = buf.readUnsignedByte();
+
+                if ((alarmDataByte1 & 0b00100000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_GEOFENCE_EXIT);
+                }
+                if ((alarmDataByte1 & 0b00000001) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_GEOFENCE_ENTER);
+                }
+
+                if ((alarmDataByte2 & 0b10000000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_DOOR);
+                }
+                if ((alarmDataByte2 & 0b00100000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_VIBRATION);
+                }
+                if ((alarmDataByte2 & 0b00001000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_POWER_OFF);
+                }
+                if ((alarmDataByte2 & 0b00000100) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_PARKING);
+                }
+                if ((alarmDataByte2 & 0b00000010) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_OVERSPEED);
+                }
+                if ((alarmDataByte2 & 0b00000001) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_SOS);
+                }
+            }
+
+            return position;
+        } else if (type == MSG_TERMINAL_ANSWER) {
+            Position position = new Position(getProtocolName());
+
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, "1" + id, id);
+            if (deviceSession == null) {
+                return null;
+            }
+
+            position.setDeviceId(deviceSession.getDeviceId());
+            position.setTime(new Date());
+
+            getLastLocation(position, null);
+
+            int mainType = buf.readUnsignedByte();
+
+            // Sub Type - Skip 1 Byte
+            buf.readUnsignedByte();
+
+            int commandExecuted = buf.readUnsignedByte();
+
+            // Reserved - Skip 2 Bytes
+            buf.readUnsignedShort();
+
+            position.set(Position.KEY_EVENT, Event.TYPE_COMMAND_RESULT);
```

Comment:
Do not do this. Event code should only include if device sends it.

Reply:
Then how would the user know that the command sent to the device was successfully executed?

I cannot create a notification / sound event if this is removed, how else to counter?

Reply:
But you asked not to hardcode KEY_RESULT, what received within the response from the device is just a byte.

Or should I use **TYPE_ENGINE_RESUME** in result?

Reply:
Well, you should use the values from the device. You should almost never hardcode any values in the decoder. I would probably use `commandExecuted` as the `KEY_RESULT` and `mainType` as the `KEY_COMMAND`. This makes the most sense from what I can tell.